### PR TITLE
Separate system config object from supervisor

### PR DESCRIPTION
**Pull request by tagomoris**
to pass system configuration to plugins

* v0.14 feature: new Plugin API expects these configuration parameters to system config
  * plugin storage path
  * default file/directory permissions
  * many others

### CODE REVIEW

File: lib/fluent/supervisor.rb [hunk 1]
```diff
@@ -585,8 +576,9 @@ def change_privilege
     end
 
     def init_engine
-      init_opts = {:suppress_interval => @suppress_interval, :suppress_config_dump => @suppress_config_dump, :without_source => @without_source}
-      Fluent::Engine.init(init_opts)
+      # init_opts = {:suppress_interval => @suppress_interval, :suppress_config_dump => @suppress_config_dump, :without_source => @without_source}
```
Comment (repeatedly) [inline lib/fluent/supervisor.rb]: This comment is needed?

File: lib/fluent/system_config.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+module Fluent
+  require 'fluent/configurable'
+
+  module SystemConfigMixin
```
Comment (repeatedly) [inline lib/fluent/system_config.rb]: Who use this module?

Comment (tagomoris) [inline lib/fluent/system_config.rb]: Plugins (or PluginSupport modules to be introduced) which need to refer system configurations.

Comment (repeatedly) [inline lib/fluent/system_config.rb]: I see

**Comment by repeatedly** [comment]
Looks good

**Comment by tagomoris** [comment]
Merged.

**Pull request merged**